inject-java/src/main/java/io/micronaut/annotation/processing/visitor/JavaClassElement.java

```diff
@@ -415,9 +404,7 @@ public AnnotationMetadata getAnnotationMetadata() {
         while (tpi.hasNext()) {
             TypeParameterElement tpe = tpi.next();
             ClassElement classElement = mirrorToClassElement(tpe.asType(), visitorContext, this.genericTypeInfo);
-            if (classElement != null) {
-                map.put(tpe.toString(), classElement);
-            }
+            map.put(tpe.toString(), classElement);
```

Comment:
This also doesn't make sense to me. If the call returns null it shouldn't be put into the map with a null value